code/modules/vehicles/mecha/mecha_ui.dm

```diff
@@ -268,6 +269,12 @@
 				tgui_alert(usr, "You cannot set a name that contains a word prohibited in IC chat!")
 				return
 			name = userinput
+		if("toggle_safety")
+			weapons_safety = !weapons_safety
```

Comment:
You should generalize `toggle safety` into a single proc that the ui and the hotkey both call, to reduce copy+paste.